src/brs/BurstGUI.java

```diff
@@ -0,0 +1,334 @@
+            }
+        } catch (Exception e) { // Catches error accessing PropertyService
+            LOGGER.error("Could not access PropertyService", e);
+            showMessage("Could not open web UI as could not read BRS configuration.");
+        }
+    }
+
+    private static void runBrs() {
+        try {
+            Burst.main(args);
+            try {
+                if (Burst.getPropertyService().getBoolean(Props.DEV_TESTNET)) {
+                    onTestNetEnabled();
+                }
+            } catch (Throwable t) {
+                LOGGER.error("Could not determine if running in testnet mode", t);
+            }
+        } catch (Throwable t) {
+            if (!(t instanceof SecurityException)) {
+                LOGGER.error("BurstGUI caught exception starting BRS", t);
+                showMessage("BurstGUI caught exception starting BRS");
+                onBrsStopped();
+            }
+        }
+    }
+
+    private static void onTestNetEnabled() {
+        stage.setTitle(stage.getTitle() + " (TESTNET)");
+        trayIcon.setToolTip(trayIcon.getToolTip() + " (TESTNET)");
+    }
+
+    private static void onBrsStopped() {
+        stage.setTitle(stage.getTitle() + " (STOPPED)");
+        trayIcon.setToolTip(trayIcon.getToolTip() + " (STOPPED)");
+    }
+
+    private static void sendJavaOutputToTextArea(TextArea textArea) {
+        System.setOut(new PrintStream(new TextAreaOutputStream(textArea, System.out)));
+        System.setErr(new PrintStream(new TextAreaOutputStream(textArea, System.err)));
+    }
+
+    private static void showMessage(String message) {
+        Platform.runLater(() -> {
+            System.out.println("Showing message: " + message);
+            Dialog dialog = new Alert(Alert.AlertType.ERROR, message, ButtonType.OK);
+            dialog.setGraphic(null);
+            dialog.setHeaderText(null);
+            dialog.setTitle("BRS Message");
+            dialog.show();
+        });
+    }
+
+    private static class TextAreaOutputStream extends OutputStream {
+        private final TextArea textArea;
+        private final PrintStream actualOutput;
+
+        private TextAreaOutputStream(TextArea textArea, PrintStream actualOutput) {
+            this.textArea = textArea;
+            this.actualOutput = actualOutput;
+        }
+
+        @Override
+        public void write(int b) {
+            writeString(new String(new byte[]{(byte)b}));
+        }
+
+        @Override
+        public void write(byte[] b) {
+            writeString(new String(b));
+        }
+
+        private void writeString(String string) {
+            actualOutput.print(string);
+            if (textArea != null) {
+                Platform.runLater(() -> textArea.appendText(string));
+            }
+        }
+    }
+
+    private static class BurstGUISecurityManager extends SecurityManager {
+
+        @Override
+        public void checkExit(int status) {
+            if (!userClosed) {
```

Comment:
personally, I try to avoid such structures. I write like this:

```
void foo(){
   if(userClosed) return;
   // ... rest here
}
```

Reply:
it's effectively the same, I prefer this way as it can be adapted better later and is more readable